modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/fun/IgniteSqlOperatorTable.java

```diff
@@ -397,6 +397,21 @@ public boolean isDeterministic() {
                     InferTypes.FIRST_KNOWN,
                     MINUS_OPERATOR_TYPES_CHECKER);
 
+    private static RelDataType deriveDatetimePlusMinusIntervalType(
+            RelDataTypeFactory typeFactory,
+            RelDataType datetimeType,
+            RelDataType intervalType,
+            boolean nullable
+    ) {
+        if (datetimeType.getSqlTypeName().allowsPrecScale(true, false)
```

Comment:
I would added an assertion that `datetimeType` is datetime indeed. The same for interval